godot-core/src/builtin/vector3.rs

```diff
@@ -85,6 +87,215 @@ impl Vector3 {
     fn to_glam(self) -> glam::Vec3 {
         glam::Vec3::new(self.x, self.y, self.z)
     }
+
+    pub fn angle_to(self, to: Self) -> f32 {
+        self.to_glam().angle_between(to.to_glam())
+    }
+
+    pub fn bezier_derivative(self, control_1: Self, control_2: Self, end: Self, t: f32) -> Self {
+        let x = bezier_derivative(self.x, control_1.x, control_2.x, end.x, t);
+        let y = bezier_derivative(self.y, control_1.y, control_2.y, end.y, t);
+        let z = bezier_derivative(self.z, control_1.z, control_2.z, end.z, t);
+
+        Self::new(x, y, z)
+    }
+
+    pub fn bezier_interpolate(self, control_1: Self, control_2: Self, end: Self, t: f32) -> Self {
+        let x = bezier_interpolate(self.x, control_1.x, control_2.x, end.x, t);
+        let y = bezier_interpolate(self.y, control_1.y, control_2.y, end.y, t);
+        let z = bezier_interpolate(self.z, control_1.z, control_2.z, end.z, t);
+
+        Self::new(x, y, z)
+    }
+
+    pub fn bounce(self, normal: Self) -> Self {
+        -self.reflect(normal)
+    }
+
+    pub fn ceil(self) -> Self {
+        Self::from_glam(self.to_glam().ceil())
+    }
+
+    pub fn clamp(self, min: Self, max: Self) -> Self {
+        Self::from_glam(self.to_glam().clamp(min.to_glam(), max.to_glam()))
+    }
+
+    pub fn cross(self, with: Self) -> Self {
+        Self::from_glam(self.to_glam().cross(with.to_glam()))
+    }
+
+    pub fn cubic_interpolate(self, b: Self, pre_a: Self, post_b: Self, weight: f32) -> Self {
+        let x = cubic_interpolate(self.x, b.x, pre_a.x, post_b.x, weight);
+        let y = cubic_interpolate(self.y, b.y, pre_a.y, post_b.y, weight);
+        let z = cubic_interpolate(self.z, b.z, pre_a.z, post_b.z, weight);
+
+        Self::new(x, y, z)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn cubic_interpolate_in_time(
+        self,
+        b: Self,
+        pre_a: Self,
+        post_b: Self,
+        weight: f32,
+        b_t: f32,
+        pre_a_t: f32,
+        post_b_t: f32,
+    ) -> Self {
+        let x = cubic_interpolate_in_time(
+            self.x, b.x, pre_a.x, post_b.x, weight, b_t, pre_a_t, post_b_t,
+        );
+        let y = cubic_interpolate_in_time(
+            self.y, b.y, pre_a.y, post_b.y, weight, b_t, pre_a_t, post_b_t,
+        );
+        let z = cubic_interpolate_in_time(
+            self.z, b.z, pre_a.z, post_b.z, weight, b_t, pre_a_t, post_b_t,
+        );
+
+        Self::new(x, y, z)
+    }
+
+    pub fn direction_to(self, to: Self) -> Self {
+        (to - self).normalized()
+    }
+
+    pub fn distance_squared_to(self, to: Self) -> f32 {
+        self.to_glam().distance_squared(to.to_glam())
+    }
+
+    pub fn distance_to(self, to: Self) -> f32 {
+        self.to_glam().distance(to.to_glam())
+    }
```

Comment:
Would be easier to just use `(to - self).length_squared()` and `(to - self).length()`.